packages/prop-controllers/src/width/width.ts

```diff
@@ -0,0 +1,135 @@
+import { LengthData, lengthDataSchema } from '../data'
+
+const widthDataSchema = lengthDataSchema
+
+export type WidthData = z.infer<typeof widthDataSchema>
+
+const responsiveWidthDataSchema = createResponsiveValueSchema(widthDataSchema)
+
+export type ResponsiveWidthData = z.infer<typeof responsiveWidthDataSchema>
+
+const widthPropControllerDataV0Schema = responsiveWidthDataSchema
+
+export type WidthPropControllerDataV0 = z.infer<
+  typeof widthPropControllerDataV0Schema
+>
+
+export const WidthPropControllerDataV1Type = 'prop-controllers::width::v1'
+
+const widthPropControllerDataV1Schema = z.object({
+  [ControlDataTypeKey]: z.literal(WidthPropControllerDataV1Type),
+  value: responsiveWidthDataSchema,
+})
+
+export type WidthPropControllerDataV1 = z.infer<
+  typeof widthPropControllerDataV1Schema
+>
+
+export const widthPropControllerDataSchema = z.union([
+  widthPropControllerDataV0Schema,
+  widthPropControllerDataV1Schema,
+])
+
+export type WidthPropControllerData = z.infer<
+  typeof widthPropControllerDataSchema
+>
+
+export const WidthPropControllerFormat = {
+  ClassName: 'makeswift::prop-controllers::width::format::class-name',
+  ResponsiveValue:
+    'makeswift::prop-controllers::width::format::responsive-value',
+} as const
+
+export type WidthPropControllerFormat =
+  typeof WidthPropControllerFormat[keyof typeof WidthPropControllerFormat]
+
+type WidthOptions = {
+  preset?: WidthPropControllerData
+  defaultValue?: LengthData
+  format?: WidthPropControllerFormat
+}
+
+type WidthDescriptorV0<
+  _T = WidthPropControllerDataV0,
+  U extends WidthOptions = WidthOptions,
+> = {
+  type: typeof Types.Width
+  options: U
+}
+
+type WidthDescriptorV1<
+  _T = WidthPropControllerData,
+  U extends WidthOptions = WidthOptions,
+> = {
+  type: typeof Types.Width
+  version: 1
+  options: U
+}
+
+export type WidthDescriptor<
+  _T = WidthPropControllerData,
+  U extends WidthOptions = WidthOptions,
+> = WidthDescriptorV0<_T, U> | WidthDescriptorV1<_T, U>
+
+export type ResolveWidthPropControllerValue<T extends WidthDescriptor> =
+  T extends WidthDescriptor
+    ? undefined extends ResolveOptions<T['options']>['format']
+      ? ResponsiveWidthData | undefined
+      : ResolveOptions<
+          T['options']
+        >['format'] extends typeof WidthPropControllerFormat.ClassName
+      ? string
+      : ResolveOptions<
+          T['options']
+        >['format'] extends typeof WidthPropControllerFormat.ResponsiveValue
+      ? ResponsiveWidthData | undefined
+      : never
+    : never
+
+/**
+ * @deprecated Imports from `@makeswift/prop-controllers` are deprecated. Use
+ * `@makeswift/runtime/controls` instead.
+ */
+export function Width<T extends WidthOptions>(
+  options: T & WidthOptions = {} as T,
+): WidthDescriptor<WidthPropControllerData, T> {
+  return { type: Types.Width, version: 1, options }
+}
+
+Width.Format = WidthPropControllerFormat
+
+export function getWidthPropControllerDataResponsiveWidthData(
```

Comment:
I think that some of this stuff is too hardcoded to what the specific control is. For example, in this case, this could be returning just responsive length data.

Reply:
Yeah, I was thinking about that too. I removed the `ResponsiveWidthData` and just used `ResponsiveLength` everywhere.